pfcli/checkpoint.py

```diff
@@ -175,6 +177,24 @@
 )
 
 
+def determine_checkpoint_status(data: Dict[str, Any]) -> None:
+    """Given the checkpoint info, inplace update the status."""
+    if data["hard_deleted"]:
+        typer.secho(
+            f"This checkpoint is hard-deleted. You cannot use this checkpoint.",
+            fg=typer.colors.RED,
+        )
+        data["status"] = "[bold red]Hard-Deleted"
+    elif data["deleted"]:
+        typer.secho(
+            f"This is a deleted checkpoint. Please restore it if you want use this.",
```

Comment:
Is it possible to know when would this checkpoint be deleted?

Reply:
@kdh0102 Changed as you requested.